akka-remote/src/main/scala/akka/remote/serialization/MiscMessageSerializer.scala

```diff
@@ -485,16 +485,16 @@ class MiscMessageSerializer(val system: ExtendedActorSystem) extends SerializerW
       a.getProtocol,
       a.getSystem,
       // technically the presence of hostname and port are guaranteed, see our serializeAddressData
-      if (a.hasHostname) Some(a.getHostname) else None,
-      if (a.hasPort) Some(a.getPort) else None)
+      a.getHostname,
+      a.getPort)
   }
   private def addressFromProto(a: ArteryControlFormats.Address): Address = {
     Address(
       a.getProtocol,
       a.getSystem,
       // technically the presence of hostname and port are guaranteed, see our serializeAddressData
-      if (a.hasHostname) Some(a.getHostname) else None,
-      if (a.hasPort) Some(a.getPort) else None)
+      a.getHostname,
+      a.getPort)
```

Comment:
is this the same thing? isn't `a.getPort` 0 when not defined and then we will use `Some(0)` instead of `None`?

Reply:
Aha, crap, you are right. 

For the record the problem here was that Scala 3 refuses to bind to the apply overload taking two options last (must be a compiler bug of some sort but didn't dig deeper into that yet).

Reply:
that constructor was private, so it looks like dotty was right about that one actually